src/gui/MainWindow.cpp

```diff
@@ -1298,15 +1308,34 @@ void MainWindow::onToggleMetronome()
 	mixer->setMetronomeActive( m_metronomeToggle->isChecked() );
 }
 
+void MainWindow::startRecordCountDown()
+{
+	if ( !m_recordCountDownTimer.isActive() )
+	{
+		m_recordCountDownTimer.start( 1000 );
+	}
+}
 
+void MainWindow::decrementRecordCountDown()
+{
+	int currentReadOut = m_countDownValue->value();
+	if ( currentReadOut - 1 < 0) {
+		m_recordCountDownTimer.stop();
+		return;
+	}
+	m_countDownValue->setValue( currentReadOut - 1 );
+}
```

Comment:
@liushuyu's choice is good. I am used to the `return` pattern; it allows to decrease indentation, which helps reading.